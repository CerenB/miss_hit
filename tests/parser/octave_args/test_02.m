% This test is extracted from the GNU Octave testsuite
% For a full copyright please refert to args.tst from GNU Octave

% one input with two possible inputs
function f (x, y)
  assert (nargin == 1);
  assert (nargout == 0);
end

function test()
 f (1);
end
