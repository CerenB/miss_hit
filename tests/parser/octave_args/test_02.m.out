----------------------------------------------------------------------
--  Parse tree for test_02.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <f>
      Inputs: Identifier <x>
      Inputs: Identifier <y>
    Body: Sequence_Of_Statements
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Identifier <nargin>
            RHS: Number_Literal <1>
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Identifier <nargout>
            RHS: Number_Literal <0>
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <test>
    Body: Sequence_Of_Statements
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <f>
          Arguments: Number_Literal <1>
----------------------------------------------------------------------
In test_02.m, line 5
| function f (x, y)
|            ^ info: this token is not linked to the ast
In test_02.m, line 5
| function f (x, y)
|              ^ info: this token is not linked to the ast
In test_02.m, line 5
| function f (x, y)
|                 ^ info: this token is not linked to the ast
In test_02.m, line 6
|   assert (nargin == 1);
|          ^ info: this token is not linked to the ast
In test_02.m, line 6
|   assert (nargin == 1);
|                      ^ info: this token is not linked to the ast
In test_02.m, line 6
|   assert (nargin == 1);
|                       ^ info: this token is not linked to the ast
In test_02.m, line 7
|   assert (nargout == 0);
|          ^ info: this token is not linked to the ast
In test_02.m, line 7
|   assert (nargout == 0);
|                       ^ info: this token is not linked to the ast
In test_02.m, line 7
|   assert (nargout == 0);
|                        ^ info: this token is not linked to the ast
In test_02.m, line 8
| end
| ^^^ info: this token is not linked to the ast
In test_02.m, line 10
| function test()
|              ^ info: this token is not linked to the ast
In test_02.m, line 10
| function test()
|               ^ info: this token is not linked to the ast
In test_02.m, line 11
|  f (1);
|    ^ info: this token is not linked to the ast
In test_02.m, line 11
|  f (1);
|      ^ info: this token is not linked to the ast
In test_02.m, line 11
|  f (1);
|       ^ info: this token is not linked to the ast
In test_02.m, line 12
| end
| ^^^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
